### DISCUSSION TITLE
btOS - BrewTarget OS GNU Linux Embedded - RaspberryPI 3

### DISCUSSION THREAD

**Discussion by cleitonbueno** [Brewtarget/brewtarget#611] [Show and tell]
**btOS - BrewTarget OS GNU Linux Embedded - RaspberryPI 3**

As a hobby and a little time, I decided to port the BrewTarget to a popular Embedded platform, today RaspberryPI 3.

<br>
<br>

- GNU Linux Embedded customized basic to BrewTarget - btOS(BrewTargetOS) 
    * GNU
    * Linux Poky 2.1.2
    * Busybox 1.24.1
    * Kernel 4.1.21
    * BrewTarget stable/2.4.0
    * Qt5.6.2

- Support Database PostgreSQL or SQLite

- Networking Lan and Wlan - Lan IP fixed 10.10.10.10/24

- Service auto init application BrewTarget after boot

- SSH Port 22
    * User: root | Pass: b33r

- SplashScreen boot custom - [**soon**]

- Enable support for touchScreen

- Image final ready to record microSD, size ~500MB with 128MB of available space, format .sdimg

<br>
<br>

**Problems**

- Problem with some icons that do not appear

<br>
<br>

**Future**

- Create git repository to keep the description of Build System for btOS(BrewTargetOS)

- Optimize and reduce boot time

- Make structure in Build System to generate btOS(BrewTargetOS) for RaspberryPI3, RaspberryPI2, Beaglebone Black and Toradex Colibri i.MX6

<br>
<br>

**Video Demo Ver. 1.00b**
https://youtu.be/jv0XjCkzsOs

<br>
<br>
Correcting some details, I can make the image 1.00b available for downloads and tests.

**Comment by rocketman768** [comment]
Love it! Good work :)

The problem with the icons is almost certainly that libqt5svg5 is not
installed. You can see most dependencies in the debian packaging:

https://github.com/Brewtarget/debian-packaging/blob/develop/debian/control

**Comment by cleitonbueno** [comment]
Hi Philip, thank you!

I added depency libqt5svg5, was the first thing I did hehe.

I'm analyzing the logs, I'm suspecting SVG support in the QtBase build.


>**\# find / -iname \"\*qt\*svg\*\"**
>/usr/include/qt5/QtSvg
>/usr/include/qt5/QtSvg/QtSvgVersion
>/usr/include/qt5/QtSvg/qtsvgversion.h
>/usr/include/qt5/QtSvg/qtsvgglobal.h
>/usr/include/qt5/QtSvg/QtSvg
>/usr/include/qt5/QtSvg/5.6.2/QtSvg
>/usr/include/qt5/QtSvg/5.6.2/QtSvg/private/qtsvgglobal_p.h
>/usr/include/qt5/QtSvg/QtSvgDepends
>/usr/lib/pkgconfig/Qt5Svg.pc
>/usr/lib/libQt5Svg.so
>/usr/lib/libQt5Svg.so.5
>/usr/lib/libQt5Svg.so.5.6
>/usr/lib/libQt5Svg.so.5.6.2
>/usr/lib/cmake/Qt5Svg
>/usr/lib/cmake/Qt5Svg/Qt5Svg_QSvgPlugin.cmake
>/usr/lib/cmake/Qt5Svg/Qt5Svg_QSvgIconPlugin.cmake
>/usr/lib/cmake/Qt5Svg/Qt5SvgConfigVersion.cmake
>/usr/lib/cmake/Qt5Svg/Qt5SvgConfig.cmake
>/usr/lib/libQt5Svg.prl
>/usr/lib/qt5/mkspecs/modules/qt_lib_svg_private.pri
>/usr/lib/qt5/mkspecs/modules/qt_lib_svg.pri

**Comment by mikfire** [comment]
And please keep me up to date on any optimizations you find. If it makes
things faster on Pi, it will be worth trying elsewhere.


Mik



--

  Mik Firestone

  mikfire@fastmail.com







On Sat, Feb 25, 2017, at 13:49, Cleiton Bueno wrote:

> Hi Philip, thank you!



> I added depency libqt5svg5, was the first thing I did hehe.



> I'm analyzing the logs, I'm suspecting SVG support in the
> QtBase build.
> — You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub[1], or mute the
> thread[2].
> 


Links:

  1. https://github.com/Brewtarget/brewtarget/issues/355#issuecomment-282503946

**Comment by pedrox86lopes** [comment]
Great!!!


*==~~==~~~==~~==~~==~~==~~==~~*

*Pedro Lopes - x86 - Perito Forense em Informática,
Grafotécnica/Documentoscopia, Judicial, Extrajudicial e Assistente técnico.*

*==~~==~~~==~~==~~==~~==~~==~~*

**Comment by cleitonbueno** [comment]
Nice!

I will correct some details and then release an image for RaspPI3.

I'm enjoying the performance, and I'm structing to port to other boards ;)

**Comment by pedrox86lopes** [comment]
Cleiton, estive pensando aqui.
Conhece o projeto CraftbeerPI?


Em 26 de fev de 2017 13:20, "Cleiton Bueno" <notifications@github.com>
escreveu:

**Comment by cleitonbueno** [comment]
Olá Pedro, não conhecia!
E achei muito legal o projeto, na verdade é mais ou menos isso que queria fazer com o brewTarget, customizar uma Distribuição Linux Embedded e deixar o mais enxuto possivel, e criar camadas para portar para outras Placas.

Como utiliza um DB, não é dificil subir um WebServer na Board RaspPI3 por exemplo e montar um mesmo esquema, e com pouco trabalho conectar um Sensor de Temperatura e alguns atuadores.

Obrigado por compartilhar.

**Comment by pedrox86lopes** [comment]
SENSACIONAL!
Vamos conversando, estou aos poucos automatizando meu processo cervejeiro
aqui em casa.
O pessoal do CraftBeer recentemente desenvolveu uma Shield pro Rasp que ja
simplifica reles e conexoes de "energia" pra ligar as panelas, enfim, vamos
nos falando.
Até!


*==~~==~~~==~~==~~==~~==~~==~~*

*Pedro Lopes - x86 - Perito Forense em Informática,
Grafotécnica/Documentoscopia, Judicial, Extrajudicial e Assistente técnico.*

*==~~==~~~==~~==~~==~~==~~==~~*


2017-02-28 23:40 GMT-03:00 Cleiton Bueno <notifications@github.com>:
